src/pickit5.c

```diff
@@ -1000,11 +1007,12 @@ static int pickit5_program_enable(const PROGRAMMER *pgm, const AVRPART *p) {
   const unsigned char *enter_prog = my.scripts.EnterProgMode;
   unsigned int enter_prog_len = my.scripts.EnterProgMode_len;
 
-  if(my.hvupdi_enabled && (my.pgm_type != PGM_TYPE_SNAP)) { // SNAP has no HV generation
+  if(my.hvupdi_enabled && can_gen_hv_pulse(my)) {   // SNAP and Basic have no HV generation
     if(p->hvupdi_variant == UPDI_ENABLE_HV_UPDI) {          // High voltage generation on UPDI line
       enter_prog = my.scripts.EnterProgModeHvSp;
       enter_prog_len = my.scripts.EnterProgModeHvSp_len;
-    } else if(p->hvupdi_variant == UPDI_ENABLE_HV_RESET) {  // High voltage generation on RST line
+    } else if(p->hvupdi_variant == UPDI_ENABLE_HV_RESET ||  // High voltage generation on RST line
+              p->hvupdi_variant == 3) {  
```

Comment:
We should probably try to define a good mnemonic for hvupdi_variant type 3. I have no idea. @xedbg what would you suggest? We have so far
https://github.com/avrdudes/avrdude/blob/8c4b69ac47c3bf5d4603ca0793b8d59af2bf310e/src/libavrdude.h#L286-L288

Reply:
Thanks - great suggestion